src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -48,47 +48,65 @@ internal static void SetSslOptions(EasyRequest easy, ClientCertificateOption cli
                 // Register the callback with libcurl.  We need to register even if there's no user-provided
                 // server callback and even if there are no client certificates, because we support verifying
                 // server certificates against more than those known to OpenSSL.
-                CURLcode answer = easy.SetSslCtxCallback(s_sslCtxCallback, userPointer);
-                switch (answer)
+                if (!CurlSslVersionDescription.StartsWith("openssl", StringComparison.OrdinalIgnoreCase) ||
+                    CurlSslVersionDescription.IndexOf("openssl/1.0", StringComparison.OrdinalIgnoreCase) != -1)
```

Comment:
The code previously was allowing other tls providers, was keeping the behavior similar to before this change. Will make it specific to openssl/1.0, just calling it out in case.